### ISSUE TITLE
Few questions 

### ISSUE THREAD

**Issue by gbougeard** [cheleb/sbt-dart-plugin#5]
hello (bonjour),
- the dartEntryPoints listed in the build.sbt file are really entry points, I mean there is no need to add all dart files they use (includes) ?

**Comment by cheleb** [comment]
Yes.

On Thu, Feb 20, 2014 at 12:20 PM, Grégory BOUGEARD <notifications@github.com

> wrote:
> 
> hello (bonjour),
> - the dartEntryPoints listed in the build.sbt file are really entry
>   points, I mean there is no need to add all dart files they use (includes) ?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cheleb/sbt-dart-plugin/issues/5
> .

## 

"Computers are useless. They can only give you answers."
- Pablo Picasso -

**Comment by gbougeard** [comment]
I have some trouble : 
I made the following tree : 
web
|--lib
     |--domain.dart
     |--component
         |--cmp
             |--my_cmp.css
             |--my_cmp.html
             |--my_cmp.dart
|--main.dart
|--opinions.html

I just added the main.dart in the entryPoints and the my_cmp.dart seems to be well compiled.
Unfortunately I'm having 404 on my_cmp.css and my_html.css

[info] application - GET /assets/packages/play_dart/component/opinion/opinion_component.css took 2ms and returned 404
[info] application - GET /assets/packages/play_dart/component/opinion/opinion_component.html took 2ms and returned 404

Do you have any idea of what could be wrong?

**Comment by cheleb** [comment]
Did you tried https://github.com/cheleb/play2dart ?

On Thu, Feb 20, 2014 at 12:43 PM, Grégory BOUGEARD <notifications@github.com

> wrote:
> 
> I have some trouble :
> I made the following tree :
> web
> |--lib
> |--domain.dart
> |--component
> |--cmp
> |--my_cmp.css
> |--my_cmp.html
> |--my_cmp.dart
> |--main.dart
> |--opinions.html
> 
> I just added the main.dart in the entryPoints and the my_cmp.dart seems to
> be well compiled.
> Unfortunately I'm having 404 on my_cmp.css and my_html.css
> 
> [info] application - GET
> /assets/packages/play_dart/component/opinion/opinion_component.css took 2ms
> and returned 404
> [info] application - GET
> /assets/packages/play_dart/component/opinion/opinion_component.html took
> 2ms and returned 404
> 
> Do you have any idea of what could be wrong?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cheleb/sbt-dart-plugin/issues/5#issuecomment-35611986
> .

## 

"Computers are useless. They can only give you answers."
- Pablo Picasso -

**Comment by gbougeard** [comment]
yep. the basic stuff works. I wanted to structure a little bit my code...

**Comment by gbougeard** [comment]
gotcha!

the trick : 
- move lib directory at root level
- import  lib content with import 'package:mypkg/myfile.dart' (if myfile.dart is in lib dir)

**Comment by cheleb** [comment]
That's perfectly right !
Dart layout is quite rigid ...
Please note the:

dartDev := true

in build.sbt this option will also deploy the dart file (to use with
dartium). This option is mandatory to work with polymer (new web_ui) and
dartium. I need to refactor the plugin to support the new "pub build".

Cheers,

Olivier

On Thu, Feb 20, 2014 at 1:47 PM, Grégory BOUGEARD
notifications@github.comwrote:

> gotcha!
> 
> the trick :
> - move lib directory at root level
> - import lib content with import 'package:mypkg/myfile.dart' (if
>   myfile.dart is in lib dir)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cheleb/sbt-dart-plugin/issues/5#issuecomment-35617831
> .

## 

"Computers are useless. They can only give you answers."
- Pablo Picasso -

**Comment by gbougeard** [comment]
I'm an absolute noob in dart, so maybe my questions are silly ;)

**Comment by cheleb** [comment]
:)
No problem

On Thu, Feb 20, 2014 at 2:07 PM, Grégory BOUGEARD
notifications@github.comwrote:

> I'm an absolute noob in dart, so maybe my questions are silly ;)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cheleb/sbt-dart-plugin/issues/5#issuecomment-35619227
> .

## 

"Computers are useless. They can only give you answers."
- Pablo Picasso -
